sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1700,6 +1700,21 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val VIEW_SCHEMA_BINDING = buildConf("spark.sql.defaultViewSchemaBinding")
+    .doc("Control the default behavior of views when the underlying schema changes. " +
+      "Valid values are: " +
+      "'BINDING': Safe casts only. " +
```

Comment:
probably we need to explain the difference between the followings:
* Safe casts
* supported casts
* Adopt type changes

Reply:
Also, 5 modes seem hard to use..Do we consider reducing it to 3?

Reply:
Reducing it to three is hard:
* UNSUPPORTED we need to switch of the feature, including dumping at DESCRIBE EXTENDED, and not storing anything fr new views.
* BINDING is the current default
* COMPENSATION is the new default we have been drifting towards (see changes to DECIMAL (done), and STRUCT (proposed)).
* That only leaves TYPE EVOLUTION vs EVOLUTION.
Slim pickings...

What we could do is, support only the first three for config.
So TYPE EVOLUTION and EVOLUTION is one you must state explicitly.
We could be even more aggressive and say: UNSUPPORTED and COMPENSATION are the only options in config.

Now, to the docs. The description on CREATE VIEW is more verbose. I could point there
The way we (don't) document configs really discourages being verbose. After all SET -v is pretty much it.

Reply:
@gengliangwang @cloud-fan 
I have reduced the config to DISABLED/COMPENSATION. That makes it de-factor a feature flag with the option to extend it to a session level default config of the need arises.